aea/aea_builder.py

```diff
@@ -1350,6 +1356,11 @@ def _load_and_add_components(
                     )
             else:
                 configuration = deepcopy(configuration)
+                configuration.update(
+                    self._custom_component_configurations.get(
+                        configuration.component_id, {}
+                    )
+                )
```

Comment:
we do the replacement only for components added from a directory, so initialization from CLI works.